### ISSUE TITLE
Uncaught TypeError: this._unregisterEvents is not a function

### ISSUE THREAD

**Issue by ilionic** [tomchentw/react-google-maps#117]
Getting this error while using `InfoBox`. 
Basically as soon as `InfoBox` is initialised and map receives new props, this error occurs.

```
Exception was thrown by user callback. TypeError: this._unregisterEvents is not a function
    at InfoBoxCreator.unregister (eval at <anonymous> (http://localhost:8888/build/app.js:5972:2), <anonymous>:20:12)
    at InfoBoxCreator.value (eval at <anonymous> (http://localhost:8888/build/app.js:5972:2), <anonymous>:48:20)
    at CallbackQueue.assign.notifyAll (eval at <anonymous> (http://localhost:8888/build/app.js:1238:2), <anonymous>:68:22)
    at ReactReconcileTransaction.ON_DOM_READY_QUEUEING.close (eval at <anonymous> (http://localhost:8888/build/app.js:1760:2), <anonymous>:81:26)
    at ReactReconcileTransaction.Mixin.closeAll (eval at <anonymous> (http://localhost:8888/build/app.js:1250:2), <anonymous>:207:25)
    at ReactReconcileTransaction.Mixin.perform (eval at <anonymous> (http://localhost:8888/build/app.js:1250:2), <anonymous>:148:16)
    at ReactUpdatesFlushTransaction.Mixin.perform (eval at <anonymous> (http://localhost:8888/build/app.js:1250:2), <anonymous>:134:20)
    at ReactUpdatesFlushTransaction.assign.perform (eval at <anonymous> (http://localhost:8888/build/app.js:1232:2), <anonymous>:95:38)
    at Object.flushBatchedUpdates (eval at <anonymous> (http://localhost:8888/build/app.js:1232:2), <anonymous>:175:19)
    at Object.ReactPerf.measure.wrapper [as flushBatchedUpdates] (eval at <anonymous> (http://localhost:8888/build/app.js:1196:2), <anonymous>:70:21)
    at ReactDefaultBatchingStrategyTransaction.Mixin.closeAll (eval at <anonymous> (http://localhost:8888/build/app.js:1250:2), <anonymous>:207:25)
    at ReactDefaultBatchingStrategyTransaction.Mixin.perform (eval at <anonymous> (http://localhost:8888/build/app.js:1250:2), <anonymous>:148:16)
    at Object.ReactDefaultBatchingStrategy.batchedUpdates (eval at <anonymous> (http://localhost:8888/build/app.js:1652:2), <anonymous>:66:19)
    at Object.enqueueUpdate (eval at <anonymous> (http://localhost:8888/build/app.js:1232:2), <anonymous>:215:22)
    at enqueueUpdate (eval at <anonymous> (http://localhost:8888/build/app.js:1220:2), <anonymous>:30:18)
    at Object.ReactUpdateQueue.enqueueSetState (eval at <anonymous> (http://localhost:8888/build/app.js:1220:2), <anonymous>:208:5) 
```

**Comment by brainkim** [comment]
Why was this closed? Also see this sometimes.

**Comment by IvoPereira** [comment]
I'm getting the same error message. Any ideas what could be causing it?

**Comment by hanigogo** [comment]
Me too. I have the same question. How to solve this problem?

**Comment by ilionic** [comment]
I was getting it quite often while using Webpack with hot module reload when new props were received.
But don't remember exactly in details. Probably better to open new issue.

**Comment by tomchentw** [comment]
Worth to reopen.

**Comment by ilyagelman** [comment]
+1 Had the same issue with Polygon and Polyline.

Disabling HMR and react-hot-loader didn't solve the issue.

**Comment by portons** [comment]
+1

**Comment by lironess** [comment]
+1

**Comment by royair** [comment]
+1

**Comment by tomchentw** [comment]
Should be fixed in [v4.8.2](https://github.com/tomchentw/react-google-maps/releases/tag/v4.8.2)
